src/shared/analytics/mixpanel.ts

```diff
@@ -0,0 +1,120 @@
+import ky from 'ky';
+import type { Account } from 'src/background/account/Account';
+import { PersistentStore } from 'src/modules/persistent-store';
+import { version } from 'src/shared/packageVersion';
+import { invariant } from '../invariant';
+import { Loglevel, logTable } from '../logger';
+import { getBaseMixpanelParams } from './shared/mixpanel-data-helpers';
+
+// TODO: move to src/env/config.ts
+const mixPanelTokenDev = 'a30959c6848ddba6ee5cb8feda61922f';
+const mixPanelTokenProd = '1713511ace475d2c78689b3d66558b62';
+const mixPanelToken =
+  process.env.NODE_ENV === 'production' ? mixPanelTokenProd : mixPanelTokenDev;
+
+class DeviceIdStore extends PersistentStore<string | undefined> {
+  constructor() {
+    super(undefined, 'deviceUUID');
+    this.ready().then(() => {
+      const value = this.getState();
+      if (!value) {
+        this.setState(crypto.randomUUID());
+      }
+    });
+  }
+
+  async getSavedState() {
+    const value = await super.getSavedState();
+    invariant(value, 'value must be generated upon initialization');
+    return value;
+  }
+}
+
+const deviceIdStore = new DeviceIdStore();
+
+class MixpanelApi {
+  baseProperties?: Record<string, unknown>;
```

Comment:
Shouldn't these properties just being set in the `getBaseMixpanelParams`?
It looks like there are 2 ways to set common params for events